Enhance InjectAgent to Handle Only HTTPGet Readiness Probes

LOADBC opened the pull request:
**What this PR does / why we need it**: 
This PR enhances the InjectAgent function to conditionally inject only HTTPGet readiness probes into the agent container. It skips unsupported readiness probe types such as exec and TCP, which are incompatible with the agent container's readiness checks. This change prevents potential errors and aligns with Kubernetes best practices for readiness probes.

**Which issue(s) this PR fixes**
This PR addresses issues where non-HTTPGet readiness probes cause the agent container to fail.
If this PR fixes an issue, include it here: https://github.com/kserve/kserve/issues/3989
Fixes #

**Type of changes**
Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

- Logs

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:

- [ ] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [x] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:
<!--  Write your release note:
1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string "action required".
2. If no release note is required, just write "NONE".
-->
```release-note

```

**Re-running failed tests**

- `/rerun-all` - rerun all failed workflows.
- `/rerun-workflow <workflow name>` - rerun a specific failed workflow. Only one workflow name can be specified. Multiple /rerun-workflow commands are allowed per comment.

hustxiayang committed:
bump to vllm0.6.2 add explicit chat template (#3964)

* explicitly give a chat template

Signed-off-by: yxia216 <yxia216@bloomberg.net>

* fix dummy model issue, fix python version smaller than 3.10, and formatting

Signed-off-by: yxia216 <yxia216@bloomberg.net>

* fix vLLMModel

Signed-off-by: yxia216 <yxia216@bloomberg.net>

* change the interface of CreateChatCompletionRequest

Signed-off-by: yxia216 <yxia216@bloomberg.net>

* update dummy model's para

Signed-off-by: yxia216 <yxia216@bloomberg.net>

* consitent with OpenAIGPTTokenizer and OpenAIGPTModel

Signed-off-by: yxia216 <yxia216@bloomberg.net>

* give a chat template if there is no

Signed-off-by: yxia216 <yxia216@bloomberg.net>

* update the response and update the readme

Signed-off-by: yxia216 <yxia216@bloomberg.net>

* update the chat_template

Signed-off-by: yxia216 <yxia216@bloomberg.net>

* update data

Signed-off-by: yxia216 <yxia216@bloomberg.net>

* add test of chat temmplate for tokenizer

Signed-off-by: yxia216 <yxia216@bloomberg.net>

* jinja2 template format

Signed-off-by: yxia216 <yxia216@bloomberg.net>

* use a simpler chat template

---------

Signed-off-by: yxia216 <yxia216@bloomberg.net>
Signed-off-by: Snehomoy <snehomoy6@gmail.com>

LOADBC committed:
Enhanced InjectAgent function to support HTTPGet readiness probes and avoid unsupported types

Signed-off-by: Snehomoy <snehomoy6@gmail.com>

spolti commented:
Hi, @LOADBC, can you please sign your commit?

spolti commented:
Can you please add a test scenario?

hustxiayang committed:
bump to vllm0.6.3 (#4001)

Signed-off-by: yxia216 <yxia216@bloomberg.net>
Signed-off-by: Snehomoy <snehomoy6@gmail.com>

LOADBC commented:
i have add the test test scenario in the `agent_injector_test.go` file the result was this : 
```
Running tool: /usr/local/go/bin/go test -timeout 30s -run ^TestInjectReadinessProbeHandling$ github.com/kserve/kserve/pkg/webhook/admission/pod

ok  	github.com/kserve/kserve/pkg/webhook/admission/pod	4.485s
```

LOADBC committed:
Update agent_injector_test.go with new test cases for readiness probes

Signed-off-by: Snehomoy <snehomoy6@gmail.com>

LOADBC commented:
@spolti Hi , could you please review the change, Thank you.

LOADBC commented:
@spolti @terrytangyuan Hi , can please help me review it.

tjandy98 committed:
Feature: Add hf transfer (#4000)

* Add hf transfer

Signed-off-by: tjandy98 <3953059+tjandy98@users.noreply.github.com>

* Add hf transfer env

Signed-off-by: tjandy98 <3953059+tjandy98@users.noreply.github.com>

---------

Signed-off-by: tjandy98 <3953059+tjandy98@users.noreply.github.com>
Signed-off-by: Snehomoy <snehomoy6@gmail.com>

sivanantha321 committed:
Fix snyk scan null error (#3974)

Signed-off-by: Sivanantham Chinnaiyan <sivanantham.chinnaiyan@ideas2it.com>
Signed-off-by: Snehomoy <snehomoy6@gmail.com>

johnugeorge committed:
Update quick install script (#4005)

Signed-off-by: Johnu George <johnugeorge109@gmail.com>
Signed-off-by: Snehomoy <snehomoy6@gmail.com>

HotsauceLee committed:
Local Model Node CR (#3978)

* init CR

Signed-off-by: Gavin Li <yli1109@bloomberg.net>

* make generate

Signed-off-by: Gavin Li <yli1109@bloomberg.net>

* make manifests

Signed-off-by: Gavin Li <yli1109@bloomberg.net>

* black format

Signed-off-by: Gavin Li <yli1109@bloomberg.net>

* fix generated python code

Signed-off-by: Gavin Li <yli1109@bloomberg.net>

* feedback

Signed-off-by: Gavin Li <yli1109@bloomberg.net>

* more feedback

Signed-off-by: Gavin Li <yli1109@bloomberg.net>

* black format

Signed-off-by: Gavin Li <yli1109@bloomberg.net>

* make manifests

Signed-off-by: Gavin Li <yli1109@bloomberg.net>

---------

Signed-off-by: Gavin Li <yli1109@bloomberg.net>
Signed-off-by: Snehomoy <snehomoy6@gmail.com>

israel-hdez committed:
Reduce E2Es dependency on CI environment (2) (#4008)

Reduce E2Es dependency on CI environment

Some code of the E2Es assume the environment is GitHub, because it is referring to GitHub-specific variables. This PR focuses on the `kserve/custom-model-grpc` container image, so that no Python code of the E2Es using this image is referencing the `github_sha` variable.

Also, a small improvement on the `get_isvc_endpoint` utility function is done to use the schema in the endpoint specified in the status of the InferenceService, rather than hard-coding to plain-text HTTP. This adds compatibility for CI environments where KServe ConfigMap has been configured with `urlScheme: https` for the Ingress.

Signed-off-by: Edgar Hernández <23639005+israel-hdez@users.noreply.github.com>
Signed-off-by: Snehomoy <snehomoy6@gmail.com>

LOADBC committed:
Fix tests by updating readiness probes to HTTPGet

Signed-off-by: Snehomoy <snehomoy6@gmail.com>

spolti commented on go.sum:
```diff
@@ -1,4 +1,3 @@
-cel.dev/expr v0.15.0/go.mod h1:TRSuuV7DlVCE/uwv5QbAiW/v8l5O8C4eEPHeu7gf7Sg=
```
not sure why all these deletions.

LOADBC replied:
Yeah sorry, The changes to `go.sum` were unintentional. They were caused by running Go tests locally, which sometimes updates the `go.sum` file to ensure dependency integrity. I've since reverted `go.sum` to its previous state and excluded it from this PR, as these changes are not related to the main purpose of this update,

spolti commented:
So, if you don't provide a HTTP get probe, the agent will have no probe?

LOADBC committed:
unintentional changes in go.sum

The changes to `go.sum` were unintentional. They were caused by running Go tests locally, which sometimes updates the `go.sum` file to ensure dependency integrity.
I've since reverted `go.sum` to its previous state and excluded it from this PR, as these changes are not related to the main purpose of this update, which focuses on "fixing the readiness probe logic."

Signed-off-by: Snehomoy <164564725+LOADBC@users.noreply.github.com>
Signed-off-by: Snehomoy <snehomoy6@gmail.com>

LOADBC committed:
Reverting back go.sum

Signed-off-by: Snehomoy <snehomoy6@gmail.com>
